compiler/rustc_mir/src/borrow_check/diagnostics/mod.rs

```diff
@@ -414,7 +414,7 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         move_prefix: &str,
     ) {
         let message = format!(
-            "{}move occurs because {} has type `{}`, which does not implement the `Copy` trait",
+            "{}`move`-borrow occurs because {} has type `{}`, which does not implement the `Copy` trait. Try to use `clone()`",
```

Comment:
Instead of mentioning `clone()` unconditionally, could you test whether `ty` implements `Clone` first?

Here's an example of how clippy tests for `Drop` traits: https://github.com/rust-lang/rust/blob/cbf666dbc1409f1afbe866ba03755e44dbd1927c/compiler/rustc_typeck/src/check/upvar.rs#L772-L779

Let me know if you want me to do a deeper review of how to do that in this situation.

Another thing we could do here is instead of having the human readable text "Try to use `.clone()`", is to use `span_suggestion` to generate a structured suggestion that rust-analyzer can apply automatically.